### PR TITLE
feat: when deserializing types, use the constructor more extensively

### PR DESCRIPTION
**Pull request by marcelgarus**
This introduces the feature discussed in #35.

### CODE REVIEW

**Comment by marcelgarus** [comment]
The test coverage CI fails only because no token from codecov.io is provided to the testing framework.

File: hive_generator/lib/src/class_builder.dart [hunk 1]
```diff
@@ -9,27 +9,57 @@ class ClassBuilder extends Builder {
...
+
+    code.writeln(')');
+
+    // There may still be fields to initialize that were not in the constructor
+    // as initializing formals. We do so using cascades.
+    for (var entry in remainingFields.entries) {
+      var field = entry.key;
+      var index = entry.value;
+      var type = fields[index].type.name;
+      code.writeln('..$field = fields[$index] as $type');
```
Comment (leisim) [inline hive_generator/lib/src/class_builder.dart]: You need to use this _cast(field.type, 'reader.read()') to cast. For example `List<dynamic>` canot be cast to `List<String>` using `as`.

File: hive_generator/lib/src/class_builder.dart [hunk 2]
```diff
@@ -9,27 +9,57 @@ class ClassBuilder extends Builder {
...
+      if (index == null) {
+        // This is a parameter of the form `this.field`, but it's not present
+        // in the binary encoding.
+        continue;
+      }
+
+      if (param.isNamed) {
+        code.write('${param.name}: ');
+      }
+      code.writeln('fields[$index] as ${param.type.name},');
```
Comment (leisim) [inline hive_generator/lib/src/class_builder.dart]: Here is the same problem as below.

**Comment by leisim** [comment]
Great, thanks a lot!

**Comment by leisim** [comment]
I'll release a new version soon.

**Pull request merged**